### PR TITLE
tree-sitter 0.17.3 (new formula)

### PR DESCRIPTION
**Pull request by dominiklohmann**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR adds tree-sitter 0.17.3 and adds it as a dependency for neovim HEAD.

See discussion in https://github.com/neovim/neovim/pull/12931 for why this is necessary and the Homebrew tap [ChimeHQ/tree-sitter](https://github.com/ChimeHQ/homebrew-tree-sitter) which I shamelessly took for inspiration.

~The strict audit fails on my machine, but I can't quite make out why it fails. Maybe someone from the Homebrew team can help me here—the PR is open for edits by maintainers.~

### CODE REVIEW

**Comment by vigoux** [comment]
While you are modifying the formula, I think the `utf8proc` dependency is not needed anymore.

File: Formula/tree-sitter.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class TreeSitter < Formula
+  desc "Parser generator tool and incremental parsing library"
+  homepage "https://tree-sitter.github.io/"
+  url "https://github.com/tree-sitter/tree-sitter/archive/0.17.3.tar.gz"
+  sha256 "a897e5c9a7ccb74271d9b20d59121d2d2e9de8b896c4d1cfaac0f8104c1ef9f8"
+  head "https://github.com/tree-sitter/tree-sitter.git"
+
+  def install
+    ENV.append_to_cflags "-mmacosx-version-min=10.10"
+    ENV.append "LDFLAGS", "-mmacosx-version-min=10.10"
```
Comment (SMillerDev) [inline Formula/tree-sitter.rb:10]: Why is this needed?

Comment (dominiklohmann) [reply Formula/tree-sitter.rb:10]: I am honestly not sure—it probably shouldn't be. It was copied from the linked repository. I'll try to remove it and will see if everything still works.

**Comment by dominiklohmann** [comment]
@SMillerDev all checks have passed, this should be good to go.

**Comment by lourenci** [comment]
Please, merge this ASAP. This is breaking [neovim's HEAD](https://github.com/neovim/neovim/issues/13216).

**Comment by SMillerDev** [comment]
We don't support HEAD builds and I don't appreciate being guilt-tripped into merging changes.

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
@SMillerDev is this good to go?

**Comment by lourenci** [comment]
@SMillerDev I didn't mean that. Sorry about that.

**Review by SMillerDev — APPROVED** [review]
